aptos-move/e2e-move-tests/src/harness.rs

```diff
@@ -342,6 +342,18 @@ impl MoveHarness {
             .new_block_with_metadata(proposer, failed_proposer_indices);
     }
 
+    // Executes the block of transactions inserting metadata at the start of the block. Returns the gas used.
+    pub fn run_block_with_metadata(
```

Comment:
What's the diff between this and new_block_with_metadata?

Reply:
`new_block_with_metadata` is a block with a single metadata txn. This also executes a block of transactions which has the form of `[block_metadata, .... all some other txns ... ]` plus returns the gas used to execute. Needed to make sure the execution runs the whole thing